addons/web_editor/static/src/js/wysiwyg/PeerToPeer.js

```diff
@@ -576,12 +586,13 @@ export class PeerToPeer {
             // hard reset: recreating a RTCPeerConnection
             if (debugShowLog)
                 console.log(
-                    `%c RTC RECOVERY: calling back client ${clientId} to salvage the connection ${pc.iceConnectionState}, reason: ${reason}`,
+                    `%c RTC RECOVERY: calling back client ${clientId} to salvage the connection ${pc.iceConnectionState} in ${backoffDelay}ms, reason: ${reason}`,
                     'background: darkorange; color: white;',
                 );
             this.removeClient(clientId);
-            await this._createClient(clientId);
-        }, delay);
+            const newClientInfos = this._createClient(clientId);
+            newClientInfos.backoffFactor = backoffFactor + 1;
+        }, backoffDelay);
```

Comment:
Shouldn't there be a cutoff too ? I mean... what's the point in trying to reconnect in one hour ? If you've been editing for one hour while someone else was editing too then tou're going to be conflicting anyway. A case could be made that it is good that we detect this conflict asap. In that case, shouldn't we reset the backoffFactor when we reconnect ? Like my tab goes to sleep, I go back to it, it should try to reconnect directly, even if the next connection attempt was scheduled in 15min due to previous backoff.

Reply:
As discussed in the office : I've added the cutoff.